File: python/tests/plugin_host_test_helper.py
# SPDX-FileCopyrightText: Copyright (c) 2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: Apache-2.0

"""Test helpers for the owned plugin-host lifecycle."""

from __future__ import annotations

from nemo_relay import JsonObject, plugin


def validate_plugin_config(config: plugin.PluginConfig | JsonObject) -> plugin.ConfigReport:
    """Validate one static plugin configuration through the unified host API."""
    return plugin.validate_exact(config)["config"]


activated_plugin_host = plugin.activate

File: crates/node/tests/plugin_host_test_helper.mjs
// SPDX-FileCopyrightText: Copyright (c) 2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import { createRequire } from 'node:module';

const require = createRequire(import.meta.url);
const plugin = require('../plugin.js');

let activation;

export function validate(config) {
  return plugin.validate(config).config;
}

export async function initialize(config, additionalPluginsToml) {
  await close();
  const nextActivation = await plugin.initialize(config, additionalPluginsToml);
  activation = nextActivation;
  return nextActivation.report.config;
}

export function report() {
  return activation?.report.config ?? null;
}

export async function close() {
  const current = activation;
  activation = undefined;
  await current?.close();
}
